warplib/src/warp_memory.cairo

```diff
@@ -14,24 +31,69 @@ impl DestructWarpMemory of Destruct::<WarpMemory> {
 }
 
 
-trait MemoryTrait {
+trait WarpMemoryTrait {
     fn initialize() -> WarpMemory;
-    fn insert(ref self: WarpMemory, position: felt252, value: felt252);
-    fn append(ref self: WarpMemory, value: felt252);
+
+    fn unsafe_write(ref self: WarpMemory, position: felt252, value: felt252);
+    fn write(ref self: WarpMemory, position: felt252, value: felt252);
+
+    fn unsafe_read(ref self: WarpMemory, position: felt252) -> felt252;
+    fn read(ref self: WarpMemory, position: felt252) -> felt252;
+
+    /// Given a certain size, it allocates the space for writing
+    fn unsafe_alloc(ref self: WarpMemory, size: felt252) -> felt252;
+    fn alloc(ref self: WarpMemory, size: felt252) -> felt252;
 }
 
 
-impl WarpMemoryImpl of MemoryTrait {
+impl WarpMemoryImpl of WarpMemoryTrait {
     fn initialize() -> WarpMemory {
         return WarpMemory {memory: Felt252DictTrait::new(), pointer: 0};
     }
 
-    fn insert(ref self: WarpMemory, position: felt252, value: felt252) {
+    fn unsafe_write(ref self: WarpMemory, position: felt252, value: felt252){
+        self.memory.insert(position, value);
+    }
+
+    fn write(ref self: WarpMemory, position: felt252, value: felt252){
+        let position_256 = u256_from_felt252(position);
+        let pointer_256 = u256_from_felt252(self.pointer);
+        if position_256 >= pointer_256 {
+           panic_with_felt252('Writing on unreserved position');
+        }
         self.memory.insert(position, value);
```

Comment:
Is optimization the reason why you don't use `unsafe_write` here?

Reply:
Yes, no need for the extra call